include/private/rdycoreimpl.h

```diff
@@ -168,17 +168,38 @@ struct _p_RDy {
   // time₋stepping solver
   TS ts;
 
-  // solution vectors (global and local)
-  Vec X, X_local;
+  // host solution vectors (global and local)
+  Vec u_global, u_local;
 
-  // residual vector
-  Vec R;
+  // host residual vector
+  Vec residual;
```

Comment:
How about we rename `residual` to `rhs`? This way it would be consistent with the name used in `ceed` struct and  we are using this vector for `TSSetRHSFunction`.